src/partition_and_pq.cpp

```diff
@@ -221,24 +225,27 @@ int generate_pq_pivots(const float *passed_train_data, size_t num_train,
       return -1;
     }
   }
-
+  
   // Calculate centroid and center the training data
   std::unique_ptr<float[]> centroid = std::make_unique<float[]>(dim);
   for (uint64_t d = 0; d < dim; d++) {
     centroid[d] = 0;
+  }
+    if (make_zero_mean) {
```

Comment:
add a comment on when we might need zero mean

Reply:
done